modules/apigee_m10n_teams/src/MonetizationTeams.php

```diff
@@ -184,4 +240,133 @@ protected function teamAccessCheck(): TeamPermissionAccessInterface {
     return \Drupal::service('apigee_m10n_teams.access_check.team_permission');
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function isLatestTermsAndConditionAccepted(string $company_id): ?bool {
+    if (!($latest_tnc = $this->getLatestTermsAndConditions())) {
+      // If there isn't a latest TnC, and there was no error, there shouldn't be
+      // anything to accept.
+      // TODO: Add a test for an org with no TnC defined.
+      return TRUE;
+    }
+    // Check the cache table.
+    if (!isset($this->companyAcceptedTermsStatus[$company_id])) {
+      // Get the latest TnC ID.
+      $latest_tnc_id = $latest_tnc->id();
+
+      // Creates a controller for getting accepted TnC.
+      $controller = $this->sdk_controller_factory->companyTermsAndConditionsController($company_id);
+
+      try {
+        $history = $controller->getTermsAndConditionsHistory();
+      }
+      catch (\Exception $e) {
+        $message = "Unable to load Terms and Conditions history for developer \n\n" . $e;
+        $this->logger->error($message);
+        throw new SdkEntityLoadException($message);
+      }
+
+      // All we care about is the latest entry for the latest TnC.
+      $latest = array_reduce($history, function ($carry, $item) use ($latest_tnc_id) {
+        /** @var \Apigee\Edge\Api\Monetization\Structure\LegalEntityTermsAndConditionsHistoryItem $item */
+        // No need to look at items other than for the current TnC.
+        if ($item->getTnc()->id() !== $latest_tnc_id) {
+          return $carry;
+        }
+        // Gets the time of the carry over item.
+        $carry_time = $carry instanceof LegalEntityTermsAndConditionsHistoryItem ? $carry->getAuditDate()->getTimestamp() : NULL;
+
+        return $item->getAuditDate()->getTimestamp() > $carry_time ? $item : $carry;
+      });
+
+      $this->companyAcceptedTermsStatus[$company_id] = ($latest instanceof LegalEntityTermsAndConditionsHistoryItem) && $latest->getAction() === 'ACCEPTED';
+    }
+
+    return $this->companyAcceptedTermsStatus[$company_id];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLatestTermsAndConditions(): ?TermsAndConditionsInterface {
```

Comment:
This looks like a copy of `\Drupal\apigee_m10n\Monetization::getLatestTermsAndConditions(). We should inject the monetization service and use that to get the latest.